shap/explainers/_tree.py

```diff
@@ -389,10 +389,8 @@ def shap_values(self, X, y=None, tree_limit=None, approximate=False, check_addit
             is used (``None``). ``-1`` means no limit.
 
         approximate : bool
-            Run fast, but only roughly approximate the Tree SHAP values. This runs a method
-            previously proposed by Saabas which only considers a single feature ordering. Take care
-            since this does not have the consistency guarantees of Shapley values and places too
-            much weight on lower splits in the tree.
+            Deprecated, will be removed in a future version. Please use the approximate argument in the
```

Comment:
about 3. how about adding `approximate` to the `__call__` and `shap_values` function so that it can be changed on different calls and is not fixed on an instance of the explainer? This way the general API is not limited to non-approximate executions.

Reply:
Yeah, that seems reasonable, you may assess whether that would have any unintended consequences (I'll do so as well, on the second pass of the code review~)

Reply:
implemented. Ready for the next round.